code/game/objects/items/storage/pouch.dm

```diff
@@ -593,9 +592,26 @@
 	else
 		return ..()
 
-/obj/item/storage/pouch/flare/full
-	fill_type = /obj/item/explosive/grenade/flare
-	fill_number = 7
+/obj/item/storage/pouch/flare/attackby_alternate(obj/item/I, mob/user, params)
+	if(!istype(I, /obj/item/weapon/gun/grenade_launcher/single_shot/flare))
+		return ..()
+	var/obj/item/weapon/gun/grenade_launcher/single_shot/flare/flare_gun = I
+	for(var/obj/item/flare in contents)
+		if(!(flare.type in can_hold))
```

Comment:
This probably isn't needed, since it shouldn't be in the pouch if it's not in can_hold.

Reply:
Fair enough.